src/FffGcodeWriter.cpp

```diff
@@ -1602,8 +1602,113 @@ void FffGcodeWriter::processSpiralizedWall(const SliceDataStorage& storage, Laye
     const bool is_top_layer = ((size_t)layer_nr == (storage.spiralize_wall_outlines.size() - 1) || storage.spiralize_wall_outlines[layer_nr + 1] == nullptr);
     ConstPolygonRef wall_outline = part.insets[0][0]; // current layer outer wall outline
     const int seam_vertex_idx = storage.spiralize_seam_vertex_indices[layer_nr]; // use pre-computed seam vertex index for current layer
+
+    std::vector<float> flows(wall_outline.size(), 1.0); // flow multiplier for each line segment
+    std::vector<Point> shifts(wall_outline.size()); // per vertex shift to compensate for line width changes
+
+    const coord_t default_line_width = mesh.settings.get<coord_t>("wall_line_width_0");
+    const double min_line_width = mesh.settings.get<Ratio>("spiralize_min_line_width");
+    const double max_line_width = mesh.settings.get<Ratio>("spiralize_max_line_width");
+
+    if (min_line_width != 1.0 || max_line_width != 1.0)
+    {
+        // wall line width can vary
+
```

Comment:
Why not use the already existing wall overlap compensation? That would save some re-implementation.

Reply:
The existing overlap compensation is horribly broken and should be avoided at all costs. Plus, I don't think it would be applicable here anyway.

Reply:
Horribly broken? What makes you say that? It's something that is done for pretty much every print. 
Might be me mistaken about it, but you're doing rather similar things, but a bit more rough (eg; checking where a line is intersecting with the model and then change its width).

Reply:
Believe me, it's horribly broken.

Reply:
I believe you. In that case we should fix this and use the fixed overlap compensation algorithm instead for this feature as well. Of course, it's unmaintainable to just keep adding new implementations every time a bug occurs in one of the existing algorithms.

Reply:
Arguably, the overlap compensation needs to be completely junked. Cura needs a new strategy for printing walls of variable widths. Just doing a sticking plaster job on the overlap compensation is not worth the effort.

Look, I don't care whether you use this PR or not but I'm not going to rewrite it to make use of the overlap compensation.